shared/tests/utility/math/TestCoordinates.cpp

```diff
@@ -0,0 +1,125 @@
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+#include <Eigen/Core>
+#include <array>
+#include <catch.hpp>
+#include <cfloat>
+#include <cmath>
+#include <fmt/format.h>
+#include <iomanip>
+#include <utility>
+#include <yaml-cpp/yaml.h>
+
+#include "utility/math/coordinates.hpp"
+#include "utility/support/yaml_expression.hpp"
+
+using utility::math::coordinates::cartesianToSpherical;
+using utility::math::coordinates::sphericalToCartesian;
+using utility::support::Expression;
+using utility::support::resolve_expression;
+
+
+/**
+ * @author Liam Craft
+ */
+
+static constexpr double ERROR_THRESHOLD = 1e-6;
+
+/** Test Data */
+static const YAML::Node test_values = YAML::LoadFile("tests/CoordinateTests.yaml");
+
+// cartesian to spherical
+SCENARIO("Cartesian coordinates can be converted to spherical coordinates", "[utility][math][coordinates]") {
+    GIVEN("A set of cartesian coordinates and a set of expected values") {
+        // cart input
+        static const std::array<Eigen::Vector3d, 200> cart_coords =
+            resolve_expression<Eigen::Vector3d, 200>(test_values["cartesianInput"]);
+        // cart to spher results
+        static const std::array<Eigen::Vector3d, 200> cart_to_spher_expected =
+            resolve_expression<Eigen::Vector3d, 200>(test_values["cartToSpher_results"]);
+        for (size_t i = 0; i < cart_coords.size(); i++) {
+            WHEN("Cartesian coordinates are converted to spherical coordinates") {
+                static const Eigen::Vector3d result = cartesianToSpherical(cart_coords.at(i));
+                THEN("The results of the calculation should be approximately equal to the expected values") {
+                    // NOTE:We use approx because floating points dont like being compared
+                    // at the epsilon value of the expected results
+                    Approx epsilon_x = Approx(cart_to_spher_expected.at(i).x()).epsilon(ERROR_THRESHOLD);
+                    Approx epsilon_y = Approx(cart_to_spher_expected.at(i).y()).epsilon(ERROR_THRESHOLD);
+                    Approx epsilon_z = Approx(cart_to_spher_expected.at(i).z()).epsilon(ERROR_THRESHOLD);
+                    // NOTE: Info message is only printed on failure
+                    INFO("Calculated x value: " << result.x());
+                    INFO("Expected x value: " << cart_to_spher_expected.at(i).x());
+                    INFO("Expected epsilon x value: " << epsilon_x.toString());
+
+                    INFO("Calculated y value: " << result.y());
+                    INFO("Expected y value: " << cart_to_spher_expected.at(i).y());
+                    INFO("Expected epsilon y value: " << epsilon_y.toString());
+
+                    INFO("Calculated z value: " << result.z());
+                    INFO("Expected z value: " << cart_to_spher_expected.at(i).z());
+                    INFO("Expected epsilon z value: " << epsilon_z.toString());
+                    INFO("Failed test value at index: " << i);
```

Comment:
Given that these can be printed on success, it's probably worth changing these `INFO`'s which reference failure as if it did fail.